### PR TITLE
TMDM-12884 [Integrated Matching] A view filtering on group size returns results from a group size minus 1

### PR DESCRIPTION
**Pull request by hw-zhu**
**What is the current behavior?** (You should also link to an open issue here)

A view filtering on group size returns results from a group size minus 1

**What is the new behavior?**

A view filtering on group size returns results from a normal group size.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/StandardQueryHandler.java [hunk 1]
```diff
@@ -1260,13 +1262,15 @@ public Criterion visit(Compare condition) {
...
-                    sqlConditionBuilder += ' ' + comparator + ' ' + value;
-                    return Restrictions.sqlRestriction(sqlConditionBuilder);
+                    StringBuilder sb = new StringBuilder();
+                    sb.append("(");
+                    sb.append("SELECT count(1) FROM");
+                    sb.append(" ").append(mainTableName).append(" ");
+                    sb.append("WHERE ").append(StorageConstants.METADATA_TASK_ID).append(" = ").append(mainTableAlias).append(".").append(StorageConstants.METADATA_TASK_ID);//$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+                    sb.append(" AND ").append(StorageConstants.METADATA_STAGING_STATUS).append(" = ").append(StagingConstants.SUCCESS_MERGE_CLUSTERS);//$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+                    sb.append(")");
+                    sb.append(" ").append(comparator).append(" ").append(value);
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/StandardQueryHandler.java]: Missing $NON-NLS-1$... for above lines?

Comment (zfkang-talend) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/StandardQueryHandler.java]: If it is a single character, it's beeter used char type, suche ' ' , ')'

Comment (hw-zhu) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/StandardQueryHandler.java]: done

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/StandardQueryHandler.java [hunk 2]
```diff
@@ -1260,13 +1262,20 @@ public Criterion visit(Compare condition) {
...
-                    sqlConditionBuilder += "where " + StorageConstants.METADATA_TASK_ID + " = " + mainTableAlias + "." + StorageConstants.METADATA_TASK_ID; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
-                    sqlConditionBuilder += ')';
-                    sqlConditionBuilder += ' ' + comparator + ' ' + value;
-                    return Restrictions.sqlRestriction(sqlConditionBuilder);
+                    StringBuilder sb = new StringBuilder();
+                    sb.append('(');//$NON-NLS-1$
+                    sb.append("SELECT count(1) FROM");//$NON-NLS-1$
+                    sb.append(' ').append(mainTableName).append(' ');//$NON-NLS-1$ //$NON-NLS-2$
+                    sb.append("WHERE ").append(StorageConstants.METADATA_TASK_ID).append('=').append(mainTableAlias).append('.').append(StorageConstants.METADATA_TASK_ID);//$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+                    if(DataRecord.CheckExistence.get()) {
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/StandardQueryHandler.java]: Format code.

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/StandardQueryHandler.java [hunk 3]
```diff
@@ -1260,13 +1262,20 @@ public Criterion visit(Compare condition) {
...
+                        sb.append(" AND ").append(StorageConstants.METADATA_STAGING_STATUS);//$NON-NLS-1$ //$NON-NLS-2$
+                        sb.append(" NOT IN (").append(StagingConstants.SUCCESS_VALIDATE).append(',');//$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+                        sb.append(StagingConstants.SUCCESS_MERGED_RECORD_TO_RESOLVE).append(',');//$NON-NLS-1$ //$NON-NLS-2$
+                        sb.append(StagingConstants.SUCCESS_MERGED_RECORD).append(')');//$NON-NLS-1$ //$NON-NLS-2$
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/StandardQueryHandler.java]: The number of NLS may not correct for above lines.

**Pull request merged**